### ISSUE TITLE
Telemetry warning shown when using Oban.Telemetry.attach_default_logger

### ISSUE THREAD

**Issue by arfl** [sorentwo/oban#578]
### Environment

- Oban Version 2.9.2
- PostgreSQL Version 13.4
- Elixir & Erlang/OTP Versions 24.0.5/1.12.3

### Current Behavior

I start `Oban.Telemetry.attach_default_logger()` in my phoenix app and get a telemetry warning:
```
[info] Function passed as a handler with ID "oban-default-logger" is local function.
This mean that it is either anonymous function or capture of function without module specified.
That may cause performance penalty when calling such handler.
For more details see note in `telemetry:attach/4` documentation.

https://hexdocs.pm/telemetry/telemetry.html#attach-4 
```

**Comment by sorentwo** [comment]
This was fixed in #563 but it isn’t released yet. A patch is coming this week.

**Comment by jeroenvisser101** [comment]
@sorentwo the same happens with oban.pro.relay:

```
[info] Function passed as a handler with ID "oban.pro.relay" is local function.
This mean that it is either anonymous function or capture of function without module specified. That may cause performance penalty when calling such handler. For more details see note in `telemetry:attach/4` documentation.
```
